cmd/minikube/cmd/service.go

```diff
@@ -67,3 +78,33 @@ func init() {
 	serviceCmd.Flags().BoolVar(&serviceURLMode, "url", false, "Display the kubernetes service URL in the CLI instead of opening it in the default browser")
 	RootCmd.AddCommand(serviceCmd)
 }
+
+func CheckService(namespace string, service string) func() error {
+	return func() error {
+		endpoints, _ := cluster.GetKubernetesEndpointsWithNamespace(namespace)
+		endpoint, _ := endpoints.Get(service)
+		if len(endpoint.Subsets) == 0 {
+			fmt.Printf("Waiting, endpoint for service:%s is not ready yet...\n", service)
+			return fmt.Errorf("Endpoint for service:%s is not ready yet\n", service)
+		}
+		for _, subset := range endpoint.Subsets {
+			if len(subset.NotReadyAddresses) != 0 {
+				fmt.Printf("Waiting, endpoint for service:%s is not ready yet...\n", service)
+				return fmt.Errorf("Endpoint for service:%s is not ready yet\n", service)
+			}
+		}
+		return nil
+	}
+}
+
+func EnsureMinikubeRunning(api libmachine.API) {
+	s, err := cluster.GetHostStatus(api)
+	if err != nil {
+		glog.Errorln("Error getting machine status:", err)
+		os.Exit(1)
+	}
+	if s != state.Running.String() {
```

Comment:
It looks like this will just return/continue if minikube isn't running. Should this os.Exit(1)?